### ISSUE TITLE
Dropout is ON for deepxde.model.Model._train_scipy?

### ISSUE THREAD

**Issue by smao-astro** [lululxvi/deepxde#68]
Hi Lu,

I was debugging my code since I was not very happy with the performance when using `L-BFGS-B`, and I noticed that there was something that I can not understand

here is a piece of my code showing I first used `Adam` and then `L-BFGS-B`

```python
    model.compile("adam", lr=0.001)
    model.train(epochs=1500, display_every=500)
    model.compile("L-BFGS-B")
```

and then the output for `L-BFGS-B` training is

> Training model...
>
> 
> Step      Train loss                                                                                    Test loss                                                                                     Test metric
> 1500      [4.55e-03, 3.91e-03, 3.73e-04, 2.27e-03, 6.69e-03, 2.55e-01, 2.32e-03, 2.09e-03, 1.68e-02]    [1.30e-03, 4.46e-03, 8.25e-05, 0.00e+00, 0.00e+00, 0.00e+00, 0.00e+00, 0.00e+00, 0.00e+00]    []  
> 2000      [3.72e-04, 1.82e-04, 1.02e-05, 2.79e-04, 2.28e-04, 1.40e-03, 5.62e-05, 8.42e-05, 5.70e-03]                                                                                                      
> 2347      [3.65e-02, 1.78e-02, 7.37e-03, 1.92e-02, 1.61e-01, 9.48e-01, 1.27e-02, 6.07e-02, 1.85e-01]    [1.42e-02, 1.48e-02, 4.06e-03, 0.00e+00, 0.00e+00, 0.00e+00, 0.00e+00, 0.00e+00, 0.00e+00]    []  

The point is at `step = 2000`, the loss is quite acceptable while the `L-BFGS-B` training quit at `step = 2347` with a loss significantly worse than `step = 2000`, which makes me confused. So could you kindly explained that:

- What determined that when `L-BFGS-B` training exits?
- Why `L-BFGS-B` training exits with a much worse even it achieved a good performance previously, does that mean I need to specify it if I what to save the model at `step = 2000`?
- While debugging this, I found the piece of code at https://github.com/lululxvi/deepxde/blob/8e811adfca060766dcbbaeec59c30300be134d00/deepxde/model.py#L246
have dropout turned on, could you explain why?

Thank you very much!

**Comment by lululxvi** [comment]
- You cam remove `L-BFGS` and use `Adam` only for lots of iterations.
- It is possible that the first step of `L-BFGS` just stuck at a bad local minimum in this case.
- It will use dropout, unless `dropout_rate` in `FNN` is set as positive.

**Comment by smao-astro** [comment]
Hi Lu, 

Thank you for your reply. 

You means "only if `dropout_rate > 0` dropout will turn on"? I see, that solved my confusion.
